test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_work_groups.cpp

```diff
@@ -0,0 +1,275 @@
+// limitations under the License.
+//
+
+#include <extensionHelpers.h>
+#include "imageHelpers.h"
+#include "mutable_command_basic.h"
+
+#include <CL/cl.h>
+#include <CL/cl_ext.h>
+
+////////////////////////////////////////////////////////////////////////////////
+// mutable dispatch tests which handle following cases:
+//
+// 1. The command buffer is created with CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR in its properties.
+// 2. The ND-range command is recorded with CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR in its properties.
+// 3. Both the command buffer and ND-range command have CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR in their properties.
+
+typedef struct {
+    const cl_command_buffer_properties_khr *command_buffer_properties;
+    const cl_ndrange_kernel_command_properties_khr *ndrange_properties;
+} Configuration;
+
+// Define the command buffer properties for each configuration
+const cl_command_buffer_properties_khr command_buffer_properties[] = {
+        CL_COMMAND_BUFFER_MUTABLE_DISPATCH_ASSERTS_KHR,
+        CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR,
+        0
+};
+
+// Define the ndrange properties
+const cl_ndrange_kernel_command_properties_khr ndrange_properties[] = {
+        CL_MUTABLE_DISPATCH_UPDATABLE_FIELDS_KHR,
+        CL_MUTABLE_DISPATCH_GLOBAL_SIZE_KHR,
+        CL_MUTABLE_DISPATCH_ASSERTS_KHR,
+        CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR,
+        0
+};
+// Initialize the array of configurations
+Configuration configurations[] = {
+        {command_buffer_properties, nullptr},
+        {nullptr,                   ndrange_properties},
+        {command_buffer_properties, ndrange_properties}
+};
+
+template<int test_case>
+struct MutableDispatchWorkGroups : public BasicMutableCommandBufferTest {
+
+    MutableDispatchWorkGroups(cl_device_id device, cl_context context,
+                              cl_command_queue queue)
+            : BasicMutableCommandBufferTest(device, context, queue), out_of_order_queue(nullptr), command_buffer(this) {
+        out_of_order_support = true;
+        config = configurations[test_case];
+    }
+
+    bool Skip() override {
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+
+        bool no_additional_wgs_support =
+                !clGetDeviceInfo(
+                        device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                        sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+                && mutable_capabilities & CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR;
+
+        bool extension_avaliable =
+                is_extension_available(device,
+                                       "cl_khr_command_buffer_mutable_dispatch")
+                == true;
+
+        return !out_of_order_support || !no_additional_wgs_support || !extension_avaliable;
+    }
+
+    cl_int SetUp(int elements) override {
```

Comment:
We should be calling base class Setup in here `BasicMutableCommandBufferTest::SetUp()`